src/modules/csm/checkpoint.py

```diff
@@ -229,18 +234,47 @@ def _prepare_committees(self, epoch: EpochNumber) -> Committees:
 
 def process_attestations(attestations: Iterable[BlockAttestation], committees: Committees) -> None:
     for attestation in attestations:
-        committee_id = (attestation.data.slot, attestation.data.index)
-        committee = committees.get(committee_id, [])
-        att_bits = _to_bits(attestation.aggregation_bits)
-        for index_in_committee, validator_duty in enumerate(committee):
-            validator_duty.included = validator_duty.included or _is_attested(att_bits, index_in_committee)
+        committee_offset = 0
+        for committee_idx in get_committee_indices(attestation):
+            committee = committees.get((attestation.data.slot, committee_idx), [])
+            att_bits = hex_bitlist_to_list(attestation.aggregation_bits)[committee_offset:][: len(committee)]
+            for index_in_committee in get_set_indices(att_bits):
+                committee[index_in_committee].included = True
+            committee_offset += len(committee)
+
+
+def get_committee_indices(attestation: BlockAttestation) -> list[CommitteeIndex]:
+    if not is_electra_attestation(attestation):
```

Comment:
could we reframe it to `if is_before_electra_attestation` ?

Reply:
I think it has the same complexity.

Reply:
semantically "not electra attestation" sounds like "any attestation from fork before and after electra" for me